### PR TITLE
adding istiod to mesh expansion gw

### PR DESCRIPTION
**Pull request by linsun**
[ ] Configuration Infrastructure
[ ] Docs
[ x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: manifests/gateways/istio-ingress/templates/meshexpansion.yaml [hunk 1]
```diff
@@ -63,6 +69,44 @@ spec:
...
+  namespace: {{ .Release.Namespace }}
+  labels:
+    release: {{ .Release.Name }}
+spec:
+  hosts:
+  - istio-pilot.{{ .Values.global.istioNamespace }}.svc.{{ .Values.global.proxy.clusterDomain }}
+  gateways:
+  - meshexpansion-gateway
+  tcp:
+  - match:
```
Comment (howardjohn) [inline manifests/gateways/istio-ingress/templates/meshexpansion.yaml]: I think you can just add another `match` section to the existing VS, but I don't think its a big difference

Comment (howardjohn) [reply manifests/gateways/istio-ingress/templates/meshexpansion.yaml]: actually is this meant to be `istio-pilot` in hosts -> `istiod` in destination?

**Review by howardjohn — COMMENTED** [review]
lgtm pending comment

File: manifests/gateways/istio-ingress/templates/meshexpansion.yaml (hunk 1 above)
Comment (linsun) [reply manifests/gateways/istio-ingress/templates/meshexpansion.yaml]: correct, fixed now, thx!

**Review by howardjohn — APPROVED** [review]

**Pull request merged**